src/dotnet-uninstall/Shared/BundleInfo/Versioning/BundleVersion.cs

```diff
@@ -51,6 +51,27 @@ public static BundleVersion FromInput<TBundleVersion>(string value)
             throw new ArgumentException();
         }
 
+        public static bool TryFromInput<TBundleVersion>(string value, out TBundleVersion version)
```

Comment:
Is that "C# allows generic type parameters to call public zero-argument constructors only"? could you give me an example?

I don't think it matters much in this case, but in general, we should not expose an internal implementation as a public API due to a language constrain. Since we barely remove any public API after release, so the cost is very high